<!DOCTYPE html>
<html class="html" lang="en-US">
 <head>

  <script type="text/javascript">
   if(typeof Muse == "undefined") window.Muse = {}; window.Muse.assets = {"required":["jquery-1.8.3.min.js", "museutils.js", "jquery.musepolyfill.bgsize.js", "jquery.watch.js", "jquery.musemenu.js", "blog---challenges-and-triumphs.css"], "outOfDate":[]};
</script>
  
  <meta http-equiv="Content-type" content="text/html;charset=UTF-8"/>
  <meta name="description" content="Articles of Both Technical Challenge and Triumph"/>
  <meta name="keywords" content="Metagenomics, sequencing, DNA, microbiology, ecology, bioinformatics"/>
  <meta name="generator" content="2015.0.0.309"/>
  <link rel="shortcut icon" href="images/a-master-favicon.ico?465361652"/>
  <title>Blog - Challenges and Triumphs</title>
  <!-- CSS -->
  <link rel="stylesheet" type="text/css" href="css/site_global.css?4052507572"/>
  <link rel="stylesheet" type="text/css" href="css/master_a-master.css?4003044962"/>
  <link rel="stylesheet" type="text/css" href="css/blog---challenges-and-triumphs.css?4156184551" id="pagesheet"/>
  <!-- Other scripts -->
  <script type="text/javascript">
   document.documentElement.className += ' js';
</script>
  <!-- JS includes -->
  <!--[if lt IE 9]>
  <script src="scripts/html5shiv.js?4241844378" type="text/javascript"></script>
  <![endif]-->
   </head>
 <body class="museBGSize">

  <div class="clearfix" id="page"><!-- column -->
   <div class="position_content" id="page_position_content">
    <div class="clearfix colelem" id="ppu1918"><!-- group -->
     <div class="clearfix grpelem" id="pu1918"><!-- group -->
      <a class="nonblock nontext clip_frame grpelem" id="u1918" href="http://www.css.cornell.edu/faculty/buckley/" target="_blank"><!-- image --><img class="block" id="u1918_img" src="images/pasted%20image%201400x934-crop-u1918.jpg" alt="" width="326" height="43"/></a>
      <a class="nonblock nontext grpelem" id="u1926-4" href="http://www.css.cornell.edu/faculty/buckley/" target="_blank"><!-- rasterized frame --><img id="u1926-4_img" alt="The Buckley Lab" width="274" height="41" src="images/u1926-4.png"/></a>
     </div>
     <div class="grpelem" id="u1170"><!-- simple frame --></div>
     <div class="clearfix grpelem" id="u1179"><!-- group -->
      <a class="nonblock nontext clip_frame grpelem" id="u1159" href="http://www.cmde.science.ubc.ca/mohn/bill.html" target="_blank"><!-- image --><img class="block" id="u1159_img" src="images/mohnlab.jpg" alt="" width="336" height="42"/></a>
      <div class="clearfix grpelem" id="u1169-4"><!-- content -->
       <p>Affiliated Websites</p>
      </div>
     </div>
    </div>
    <div class="clearfix colelem" id="pu1668"><!-- group -->
     <div class="clearfix grpelem" id="u1668"><!-- group -->
      <div class="clearfix grpelem" id="u1557"><!-- group -->
       <div class="grpelem" id="u1667"><!-- simple frame --></div>
      </div>
     </div>
     <div class="clearfix grpelem" id="u1556"><!-- column -->
      <div class="clearfix colelem" id="u1554-4"><!-- content -->
       <p id="u1554-2">Previous Posts</p>
      </div>
      <nav class="MenuBar clearfix colelem" id="menuu1506"><!-- vertical box -->
       <div class="MenuItemContainer clearfix colelem" id="u1547"><!-- horizontal box -->
        <div class="MenuItem MenuItemWithSubMenu clearfix grpelem" id="u1548"><!-- horizontal box -->
         <div class="MenuItemLabel clearfix grpelem" id="u1550-6"><!-- content -->
          <p id="u1550-2">Nextera Illumina Read</p>
          <p id="u1550-4">Cleaning</p>
         </div>
        </div>
       </div>
       <div class="MenuItemContainer clearfix colelem" id="u1660"><!-- horizontal box -->
        <a class="nonblock nontext MenuItem MenuItemWithSubMenu clearfix grpelem" id="u1663" href="index.html"><!-- horizontal box --><div class="MenuItemLabel clearfix grpelem" id="u1666-4"><!-- content --><p id="u1666-2">HOME</p></div></a>
       </div>
      </nav>
     </div>
     <div class="clearfix grpelem" id="u1558-4"><!-- content -->
      <p id="u1558-2"><span id="u1558">Suspicious K-mer Content at Start of Illumina Reads Generated Using Nextera Library Kits</span></p>
     </div>
    </div>
    <div class="clearfix colelem" id="pu1560-4"><!-- group -->
     <div class="clearfix grpelem" id="u1560-4"><!-- content -->
      <p>Transposon insertion bias or a different sort of sequencing artifact?</p>
     </div>
     <div class="clearfix grpelem" id="u1561-4"><!-- content -->
      <p>May 2nd, 2014</p>
     </div>
    </div>
    <div class="clearfix colelem" id="pu1559-151"><!-- group -->
     <div class="clearfix grpelem" id="u1559-151"><!-- content -->
      <p id="u1559-11">Many researchers have reported some form of biased representation of k-mer content at the beginning (5'-end) of sequencing data generated from Illumina library prep kits. Here are two threads with discussion of this phenomenon for Nextera kits (<a class="nonblock" href="http://seqanswers.com/forums/showthread.php?t=20561">HERE</a>) and TruSeq kits (<a class="nonblock" href="http://seqanswers.com/forums/showthread.php?t=11843">HERE</a>). The general consensus is that that the patterns of bias are produced by modest and uncontrollable preferential interactions between certain nucleic acid sequences and the random hexamer PCR primers or transposable elements (Here's <a class="nonblock" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3292447/">a paper</a> demonstrating such a bias for the Tn5 transposable element related to the one used in the Nextera kit).</p>
      <p id="u1559-12">&nbsp;</p>
      <p id="u1559-13">&nbsp;</p>
      <p id="u1559-14">&nbsp;</p>
      <p id="u1559-15">&nbsp;</p>
      <p id="u1559-16">&nbsp;</p>
      <p id="u1559-17">&nbsp;</p>
      <p id="u1559-18">&nbsp;</p>
      <p id="u1559-19">&nbsp;</p>
      <p id="u1559-20">&nbsp;</p>
      <p id="u1559-21">&nbsp;</p>
      <p id="u1559-22">&nbsp;</p>
      <p id="u1559-23">&nbsp;</p>
      <p id="u1559-24">&nbsp;</p>
      <p id="u1559-25">&nbsp;</p>
      <p id="u1559-26">&nbsp;</p>
      <p id="u1559-27">&nbsp;</p>
      <p id="u1559-28">&nbsp;</p>
      <p id="u1559-29">&nbsp;</p>
      <p id="u1559-30">&nbsp;</p>
      <p id="u1559-31">&nbsp;</p>
      <p id="u1559-32">&nbsp;</p>
      <p id="u1559-33">&nbsp;</p>
      <p id="u1559-34">&nbsp;</p>
      <p id="u1559-39">The main question is: &quot;What kind of effect might this have on assembly?&quot; The feedback from a number of researchers was that they observed no significant differences in assembly based on whether they trimming these features (i.e. the first 14bp of affected reads) or not. For <a class="nonblock" href="http://microsizedmind.wordpress.com/2014/02/21/is-improvement-possible-quality-control-of-a-illumina-nextera-dataset-for-the-novo-genome-assembly/">one of those researchers</a>, however, the lack of improvement meant abandoning his data. Interestingly, the problems presented by this unexplained k-mer pattern is not readily addressed using trimming software, which hunt and prey on adapter sequences. This isn't surprising given the fact the k-mers sequences weren't found in the sequencing adapters or indices (see the list of the top most abundant repeats re-constructed from the k-mer chart left of figure 1).</p>
      <p id="u1559-47">NOTE: One more readily solved, but different problem, is trimming reads which had such short fragments that the sequenced read included the opposing paired end adapter. This can be remedied using trimming software (like <a class="nonblock" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>) and is described <a class="nonblock" href="http://pathogenomics.bham.ac.uk/blog/2013/04/adaptor-trim-or-die-experiences-with-nextera-libraries/">HERE</a> (@ pathogenomenick).</p>
      <p id="u1559-48">&nbsp;</p>
      <p id="u1559-53">So, if I haven't lost you by assuaging your worries by saying: &quot;some people just don't worry about it&quot;, I'll explain what prompted me to write this piece. I have become skeptical about the explanation based on transposon bias due to the uncanny consistency in which these patterns of bias materialize across different samples and projects.&nbsp; All eight of the samples I prepared (and sequenced over two separate runs) had the same identical pattern, despite the expectation of a very different nucleic acid composition as the libraries were made from complex soil microbial communities. My suspicions grew when I began searching for explanations and found identical, or nearly identical, <a class="nonblock" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> output graphs from four other researchers from related posts in forums and blogs.</p>
      <p id="u1559-54">&nbsp;</p>
      <p id="u1559-55">&nbsp;</p>
      <p id="u1559-56">&nbsp;</p>
      <p id="u1559-57">&nbsp;</p>
      <p id="u1559-58">&nbsp;</p>
      <p id="u1559-59">&nbsp;</p>
      <p id="u1559-60">&nbsp;</p>
      <p id="u1559-61">&nbsp;</p>
      <p id="u1559-62">&nbsp;</p>
      <p id="u1559-63">&nbsp;</p>
      <p id="u1559-64">&nbsp;</p>
      <p id="u1559-65">&nbsp;</p>
      <p id="u1559-66">&nbsp;</p>
      <p id="u1559-67">&nbsp;</p>
      <p id="u1559-68">&nbsp;</p>
      <p id="u1559-69">&nbsp;</p>
      <p id="u1559-70">&nbsp;</p>
      <p id="u1559-71">&nbsp;</p>
      <p id="u1559-72">&nbsp;</p>
      <p id="u1559-73">&nbsp;</p>
      <p id="u1559-74">&nbsp;</p>
      <p id="u1559-75">&nbsp;</p>
      <p id="u1559-76">&nbsp;</p>
      <p id="u1559-77">&nbsp;</p>
      <p id="u1559-78">&nbsp;</p>
      <p id="u1559-83">In my opinion, even if transposons had ultra-specific target sites in downstream locations from where the adapter sequence gets inserted, there is a low probability of getting such clean and comparable data by accident. I may be jaded by how few of my experiments ever look so clean, but I would be surprised we just accidentally nailed down the biased recognition sites of whatever variant of tn5 transposon the Nextera people are using. The more likely explanation would involve something to do with the manufacturing process of the Nextera kits, whether it be a characteristic of how the adapters or primers are synthesized or part of the <a class="nonblock" href="http://www.cambio.co.uk/library/images/html_images/nextera_illumina_fig_thumb.gif">transposase sequence itself</a> being inserted withing the read. I admit to not having an answer, but I hope this post might spark an more complete explanation. I have contacted Illumina and will report back their answer.</p>
      <p id="u1559-84">&nbsp;</p>
      <p id="u1559-89">In the meantime, PLEASE COMMENT @ the SEQanswers Thread Addressing This Question Started <a class="nonblock" href="http://seqanswers.com/forums/showthread.php?t=17219">HERE</a>.</p>
      <p id="u1559-90">&nbsp;</p>
      <p id="u1559-91">&nbsp;</p>
      <p id="u1559-93">-- UPDATE --</p>
      <p id="u1559-95"><span id="u1559-94">(May, 7th, 2014)</span></p>
      <p id="u1559-96">&nbsp;</p>
      <p id="u1559-104">This post generated some good discussion on the <a class="nonblock" href="http://seqanswers.com/forums/showthread.php?t=17219">SEQanswers thread</a>, and both the Illumina rep. and a thoughtful contributer in the thread referenced <a class="nonblock" href="http://genomebiology.com/2010/11/12/r119">the following paper</a> which compared different library preparation methods to the &quot;tagmentation&quot; with transposases. They detected bias as a result of the native Tn5 transposase recognition site: AGNTYWRANCT (where N is any nucleotide, R is A or G, W is A or T, and Y is C or T).&nbsp; When I search through my reads based on the 6 base k-mers I reconstruct from the table below, I DO find similarity to the reported recognition site. The similarity is not perfect, but this may be due to the fact the transposase used in the Nextera kit is not the same as what was used in the paper.</p>
      <p id="u1559-105">&nbsp;</p>
      <p id="u1559-106">&nbsp;</p>
      <p id="u1559-107">&nbsp;</p>
      <p id="u1559-108">&nbsp;</p>
      <p id="u1559-109">&nbsp;</p>
      <p id="u1559-111">Now, with the actual sequence information related to the transposon bias in hand, I'm much more comfortable with that explanation. And, when I look at the total number of sequences which contain one of these k-mers in the first 14 bases of sequence, I find that the counts are really quite small (0.3%). My initial fears were based on an incorrect calculation where I summed the &quot;count&quot; column in the FastQC output for the seven major over-represented k-mers and divided by the total number of sequences. When I was reviewing this, I realized that the count data provided by FastQC is the total number of occurrences anywhere in the read and was, therefore, a vast overestimate of the reads with disproportionate abundances at the start of the read.</p>
      <p id="u1559-113">One additional finding, which further negates the importance of this bias is that when I randomly BLAST the first 14 bp of reads containing the repeats, I do not see a consistent taxonomic signal.</p>
      <p id="u1559-114">&nbsp;</p>
      <p id="u1559-116">I thank everyone for sharing information and I hope this may have helped some newbies, like myself, in assessing the quality of their sequencing data. I am preparing to assemble my metagenomes; I will make one final brief update regarding the success of the assembly to finish this post.</p>
      <p id="u1559-117">&nbsp;</p>
      <p id="u1559-118">&nbsp;</p>
      <p id="u1559-119">&nbsp;</p>
      <p id="u1559-120">&nbsp;</p>
      <p id="u1559-121">&nbsp;</p>
      <p id="u1559-122">&nbsp;</p>
      <p id="u1559-123">&nbsp;</p>
      <p id="u1559-124">&nbsp;</p>
      <p id="u1559-125">&nbsp;</p>
      <p id="u1559-126">&nbsp;</p>
      <p id="u1559-127">&nbsp;</p>
      <p id="u1559-128">&nbsp;</p>
      <p id="u1559-129">&nbsp;</p>
      <p id="u1559-130">&nbsp;</p>
      <p id="u1559-131">&nbsp;</p>
      <p id="u1559-132">&nbsp;</p>
      <p id="u1559-133">&nbsp;</p>
      <p id="u1559-134">&nbsp;</p>
      <p id="u1559-135">&nbsp;</p>
      <p id="u1559-136">&nbsp;</p>
      <p id="u1559-137">&nbsp;</p>
      <p id="u1559-138">&nbsp;</p>
      <p id="u1559-139">&nbsp;</p>
      <p id="u1559-140">&nbsp;</p>
      <p id="u1559-141">&nbsp;</p>
      <p id="u1559-142">&nbsp;</p>
      <p id="u1559-143">&nbsp;</p>
      <p id="u1559-144">&nbsp;</p>
      <p id="u1559-145">&nbsp;</p>
      <p id="u1559-147">- Roli Wilhelm</p>
      <p>&nbsp;</p>
      <p>&nbsp;</p>
     </div>
     <div class="clearfix grpelem" id="u1574"><!-- column -->
      <div class="clearfix colelem" id="pu1573-19"><!-- group -->
       <div class="clearfix grpelem" id="u1573-19"><!-- content -->
        <p id="u1573-2">List of Top Repeating Oligomers</p>
        <p id="u1573-4">ATCGAT</p>
        <p id="u1573-6">CTCGAG</p>
        <p id="u1573-8">TGATCA</p>
        <p id="u1573-10">CGATCG</p>
        <p id="u1573-12">CTGCTG</p>
        <p id="u1573-14">GTCGAC</p>
        <p id="u1573-16">CAGCAG</p>
        <p>&nbsp;</p>
       </div>
       <a class="nonblock nontext clip_frame grpelem" id="u1567" href="fastqc-graph-c12.html"><!-- image --><img class="block" id="u1567_img" src="images/skeptical461x286.jpg" alt="" width="460" height="285"/></a>
      </div>
      <div class="clearfix colelem" id="u1572-4"><!-- content -->
       <p>My Own Example of K-mer Bias From 100bp Paired-end Reads from a HiSeq Run. (Click to Enlarge)</p>
      </div>
     </div>
     <div class="clearfix grpelem" id="u1600"><!-- column -->
      <a class="nonblock nontext clip_frame colelem" id="u1594" href="http://pathogenomics.bham.ac.uk/blog/wp-content/uploads/pretrimmed_nuccomp.png"><!-- image --><img class="block" id="u1594_img" src="images/pretrimmed_nuccomp.jpg" alt="" width="475" height="299"/></a>
      <div class="clearfix colelem" id="u1599-13"><!-- content -->
       <p>An identical FastQC output from an entirely different sample and sequencing runs using the Nextera prep kit, with other nearly identical graphs <a class="nonblock" href="http://microsizedmind.files.wordpress.com/2014/02/fastqc_phalo_raw_data_gc_content.png?w=640&amp;h=261">HERE</a>, <a class="nonblock" href="https://pbs.twimg.com/media/BfuVswZCAAAV-jP.png">HERE</a> and <a class="nonblock" href="http://seqanswers.com/forums/attachment.php?attachmentid=3011&amp;d=1398861581">HERE</a>.</p>
      </div>
     </div>
     <div class="grpelem" id="u1650"><!-- custom html -->
      
<a href="https://twitter.com/rabbleroyal" class="twitter-follow-button" data-lang="en" data-show-screen-name="true" data-size="medium"></a>

</div>
     <div class="clearfix grpelem" id="u1698"><!-- column -->
      <div class="clearfix colelem" id="pu1694-9"><!-- group -->
       <div class="clearfix grpelem" id="u1694-9"><!-- content -->
        <p>CG<span id="u1694-2">AGATC</span>G<span id="u1694-4">A</span>T<span id="u1694-6">C</span></p>
       </div>
       <div class="clearfix grpelem" id="u1697-4"><!-- content -->
        <p>recurring sequence in reads</p>
       </div>
      </div>
      <div class="clearfix colelem" id="pu1695-9"><!-- group -->
       <div class="clearfix grpelem" id="u1695-9"><!-- content -->
        <p><span id="u1695">AGNTY</span>W<span id="u1695-3">R</span>A<span id="u1695-5">N</span>CT</p>
       </div>
       <div class="clearfix grpelem" id="u1696-4"><!-- content -->
        <p>transposase recognition site</p>
       </div>
      </div>
     </div>
     <div class="clearfix grpelem" id="u1751"><!-- column -->
      <div class="clearfix colelem" id="u1678-4"><!-- content -->
       <p>Click to Enlarge</p>
      </div>
      <a class="nonblock nontext clip_frame colelem" id="u1673" href="fastqc-k-mer-table.html"><!-- image --><img class="block" id="u1673_img" src="images/top%20k-mers569x340.jpg" alt="" width="568" height="340"/></a>
     </div>
    </div>
    <div class="verticalspacer"></div>
    <div class="clearfix colelem" id="u1236-7"><!-- content -->
     <p><span id="u1236">Last updated </span> <span id="u1236-3">Jan. 26, 2017</span></p>
    </div>
   </div>
  </div>
  <!-- JS includes -->
  <script type="text/javascript">
   if (document.location.protocol != 'https:') document.write('\x3Cscript src="http://musecdn.businesscatalyst.com/scripts/4.0/jquery-1.8.3.min.js" type="text/javascript">\x3C/script>');
</script>
  <script type="text/javascript">
   window.jQuery || document.write('\x3Cscript src="scripts/jquery-1.8.3.min.js" type="text/javascript">\x3C/script>');
</script>
  <script src="scripts/museutils.js?183364071" type="text/javascript"></script>
  <script src="scripts/jquery.musepolyfill.bgsize.js?4004268962" type="text/javascript"></script>
  <script src="scripts/jquery.musemenu.js?3957776250" type="text/javascript"></script>
  <script src="scripts/jquery.watch.js?71412426" type="text/javascript"></script>
  <!-- Other scripts -->
  <script type="text/javascript">
   $(document).ready(function() { try {
(function(){var a={},b=function(a){if(a.match(/^rgb/))return a=a.replace(/\s+/g,"").match(/([\d\,]+)/gi)[0].split(","),(parseInt(a[0])<<16)+(parseInt(a[1])<<8)+parseInt(a[2]);if(a.match(/^\#/))return parseInt(a.substr(1),16);return 0};(function(){$('link[type="text/css"]').each(function(){var b=($(this).attr("href")||"").match(/\/?css\/([\w\-]+\.css)\?(\d+)/);b&&b[1]&&b[2]&&(a[b[1]]=b[2])})})();(function(){$("body").append('<div class="version" style="display:none; width:1px; height:1px;"></div>');
for(var c=$(".version"),d=0;d<Muse.assets.required.length;){var f=Muse.assets.required[d],g=f.match(/([\w\-\.]+)\.(\w+)$/),k=g&&g[1]?g[1]:null,g=g&&g[2]?g[2]:null;switch(g.toLowerCase()){case "css":k=k.replace(/\W/gi,"_").replace(/^([^a-z])/gi,"_$1");c.addClass(k);var g=b(c.css("color")),h=b(c.css("background-color"));g!=0||h!=0?(Muse.assets.required.splice(d,1),"undefined"!=typeof a[f]&&(g!=a[f]>>>24||h!=(a[f]&16777215))&&Muse.assets.outOfDate.push(f)):d++;c.removeClass(k);break;case "js":k.match(/^jquery-[\d\.]+/gi)&&
typeof $!="undefined"?Muse.assets.required.splice(d,1):d++;break;default:throw Error("Unsupported file type: "+g);}}c.remove();if(Muse.assets.outOfDate.length||Muse.assets.required.length)c="Some files on the server may be missing or incorrect. Clear browser cache and try again. If the problem persists please contact website author.",(d=location&&location.search&&location.search.match&&location.search.match(/muse_debug/gi))&&Muse.assets.outOfDate.length&&(c+="\nOut of date: "+Muse.assets.outOfDate.join(",")),d&&Muse.assets.required.length&&(c+="\nMissing: "+Muse.assets.required.join(",")),alert(c)})()})();/* body */
Muse.Utils.transformMarkupToFixBrowserProblemsPreInit();/* body */
Muse.Utils.prepHyperlinks(true);/* body */
Muse.Utils.initWidget('.MenuBar', function(elem) { return $(elem).museMenu(); });/* unifiedNavBar */
Muse.Utils.fullPage('#page');/* 100% height page */
Muse.Utils.showWidgetsWhenReady();/* body */
Muse.Utils.transformMarkupToFixBrowserProblems();/* body */
} catch(e) { if (e && 'function' == typeof e.notify) e.notify(); else Muse.Assert.fail('Error calling selector function:' + e); }});
</script>
  
  <!--HTML Widget code-->
  
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="https://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

   </body>
</html>
